numpy/ma/core.py

```diff
@@ -6206,16 +6227,23 @@ def __str__(self):
         return str(masked_print_option._display)
 
     def __repr__(self):
-        return 'masked'
-
-    def flatten(self):
-        return masked_array([self._data], dtype=float, mask=[True])
+        if self is masked:
+            return 'masked'
+        else:
+            # something is wrong, make it obvious
```

Comment:
Just to be sure, this will not happen in any normal usage, correct? (e.g., from the test below i assume `.copy` returns something else)